File: tests/seleniumTest.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Initialize the WebDriver (e.g., ChromeDriver)
driver = webdriver.Chrome()

# Navigate to your React application's login page
driver.get("http://localhost:3000/login")

# Wait for the email input to be visible
WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.ID, "login-form_email")))

# Find elements
email_input = driver.find_element(By.ID, "login-form_email")
password_input = driver.find_element(By.ID, "login-form_password")
login_button = driver.find_element(By.CLASS_NAME, "login-button")

# Fill out the login form
email_input.send_keys("joashlwj@gmail.com")
password_input.send_keys("Password@12345")
login_button.click()

# Wait for navigation to the account page
WebDriverWait(driver, 10).until(EC.url_contains("/account"))

# Check if the login was successful
assert "/account" in driver.current_url

# Close the WebDriver
driver.quit()

File: client/src/tests/seleniumTest.js
const { Builder, By, Key, until } = require("selenium-webdriver");
const chrome = require("selenium-webdriver/chrome");

(async () => {
  // Set up Chrome options
  const options = new chrome.Options();
  options.addArguments("--headless");

  // Initialize the WebDriver
  const driver = await new Builder()
    .forBrowser("chrome")
    .setChromeOptions(options)
    .build();

  try {
    // Navigate to your React application's login page
    await driver.get("https://gracious-kare.cloud/login");

    // Wait for the email input to be visible
    await driver.wait(until.elementLocated(By.id("login-form_email")), 10000);

    // Find elements
    const emailInput = await driver.findElement(By.id("login-form_email"));
    const passwordInput = await driver.findElement(
      By.id("login-form_password")
    );
    const loginButton = await driver.findElement(By.className("login-button"));

    // Fill out the login form
    await emailInput.sendKeys("joashlwj@gmail.com");
    await passwordInput.sendKeys("Password@12345");
    await loginButton.click();

    // Wait for navigation to the account page
    await driver.wait(until.urlContains("/account"), 10000);

    // Check if the login was successful
    const currentUrl = await driver.getCurrentUrl();
    if (!currentUrl.includes("/account")) {
      throw new Error("Login was not successful");
    }
  } finally {
    // Close the WebDriver
    await driver.quit();
  }
})();
